lib/Listeners/FilesSharingLoadAdditionalScriptsListener.php

```diff
@@ -25,13 +25,14 @@
 
 namespace OCA\Text\Listeners;
 
+use OCA\Files_Sharing\Event\BeforeTemplateRenderedEvent;
 use OCA\Text\Service\InitialStateProvider;
 use OCP\EventDispatcher\Event;
 use OCP\EventDispatcher\IEventListener;
 use OCP\IConfig;
 use OCP\Util;
 
-/** @implements IEventListener<Event> */
+/** @implements IEventListener<BeforeTemplateRenderedEvent> */
```

Comment:
```suggestion
/** @implements IEventListener<Event|BeforeTemplateRenderedEvent> */
```

This and a stub for the BeforeTemplateRenderedEvent like in https://github.com/nextcloud/text/blob/main/tests/stub.phpstub#L13-L15 should work